### ISSUE TITLE
`overlayAccessibilityLabel` prop for Dialog component

### ISSUE THREAD

**Issue by meatnordrink** [callstack/react-native-paper#3911]
**Is your feature request related to a problem? Please describe.**
The accessibility label for modals is editable via `overlayAccessibilityLabel`, which is necessary given the accessibility difficulties surrounding modals; but there is no corresponding prop on dialogs, leaving us unable to edit the default "close modal" text announced to screen readers.

**Describe the solution you'd like**
I would like to see an `overlayAccessibilityLabel` prop on `Dialog` components.

**Describe alternatives you've considered**
If some of the accessibility props of RN were exposed through `Dialog` (like `accessibilityLabel`, etc.), that might be an alternative.

**Additional context**
n/a
